virtualizarr/translators/kerchunk.py

```diff
@@ -93,32 +94,31 @@ def from_kerchunk_refs(decoded_arr_refs_zarray) -> "ArrayV3Metadata":
     )
 
 
-def virtual_vars_and_metadata_from_kerchunk_refs(
-    vds_refs: KerchunkStoreRefs,
-    drop_variables: list[str] | None = None,
+def manifeststore_from_kerchunk_refs(
+    refs: KerchunkStoreRefs,
+    group: str | None = None,
     fs_root: str | None = None,
-) -> tuple[Mapping[str, Variable], dict[str, Any], list[str]]:
-    """
-    Parses all useful information from a set kerchunk references (for a single group).
+) -> ManifestStore:
+    # both group=None and group='' mean to read root group
+    if group:
+        refs = extract_group(refs, group)
+
+    arr_names = find_var_names(refs)
+    # TODO do we need drop_variables here?
+
+    # TODO support iterating over multiple nested groups
+    marrs = {
+        arr_name: manifestarray_from_kerchunk_refs(refs, arr_name, fs_root=fs_root)
+        for arr_name in arr_names
+    }
 
-    Parameters
-    ----------
-    drop_variables
-        Variables in the file to not bother generating chunk metadata for.
-    fs_root
-        The root of the fsspec filesystem on which these references were generated.
-        Required if any paths are relative in order to turn them into absolute paths (which virtualizarr requires).
-    """
+    # TODO probably need to parse the group-level attributes more here
+    attributes = fully_decode_arr_refs(refs["refs"]).get(".zattrs", {})
 
-    virtual_vars = virtual_vars_from_kerchunk_refs(
-        vds_refs,
-        drop_variables=drop_variables,
-        fs_root=fs_root,
-    )
-    ds_attrs = fully_decode_arr_refs(vds_refs["refs"]).get(".zattrs", {})
-    coord_names = ds_attrs.pop("coordinates", [])
+    manifestgroup = ManifestGroup(arrays=marrs, attributes=attributes)
 
-    return virtual_vars, ds_attrs, coord_names
+    # TODO what should the obstore store be?
+    return ManifestStore(group=manifestgroup)
```

Comment:
@maxrjones I think I need `default_object_store()` in order to continue with this PR.